apps/worker/src/request_processor/dsnp.announcement.processor.ts

```diff
@@ -0,0 +1,338 @@
+                hash: [hashedContent],
+                height: reference.height,
+                width: reference.width,
+                type: 'Link',
+                href: await this.formIpfsUrl(reference.referenceId),
+              };
+              imageLinks.push(image);
+            });
+            const imageActivity: ActivityContentImage = {
+              type: 'Image',
+              name: asset.name || '',
+              url: imageLinks,
+            };
+
+            attachments.push(imageActivity);
+            break;
+          }
+          case AttachmentTypeDto.VIDEO: {
+            const videoLinks: ActivityContentVideoLink[] = [];
+            let duration = '';
+            asset.references?.forEach(async (reference) => {
+              const contentBuffer = await this.ipfsService.getPinned(reference.referenceId);
+              const hashedContent = await calculateDsnpHash(contentBuffer);
+              const video: ActivityContentVideoLink = {
+                mediaType: reference.mimeType,
+                hash: [hashedContent],
+                height: reference.height,
+                width: reference.width,
+                type: 'Link',
+                href: await this.formIpfsUrl(reference.referenceId),
+              };
+              duration = duration ?? reference.duration ?? '';
+              videoLinks.push(video);
+            });
+            const videoActivity: ActivityContentVideo = {
+              type: 'Video',
+              name: asset.name || '',
+              url: videoLinks,
+              duration,
+            };
+
+            attachments.push(videoActivity);
+            break;
+          }
+          case AttachmentTypeDto.AUDIO: {
+            const audioLinks: ActivityContentAudioLink[] = [];
+            let duration = '';
+            asset.references?.forEach(async (reference) => {
+              const contentBuffer = await this.ipfsService.getPinned(reference.referenceId);
+              const hashedContent = await calculateDsnpHash(contentBuffer);
+              duration = duration ?? reference.duration ?? '';
+              const audio: ActivityContentAudioLink = {
+                mediaType: reference.mimeType,
+                hash: [hashedContent],
+                type: 'Link',
+                href: await this.formIpfsUrl(reference.referenceId),
+              };
+              audioLinks.push(audio);
+            });
+            const audioActivity: ActivityContentAudio = {
+              type: 'Audio',
+              name: asset.name || '',
+              url: audioLinks,
+              duration,
+            };
+
+            attachments.push(audioActivity);
+            break;
+          }
+          default:
+            throw new Error(`Unsupported attachment type ${typeof asset.type}`);
+        }
+      });
+    }
+
+    const note = createNote(noteContent?.content.content ?? '', new Date(noteContent?.content.published ?? ''), {
```

Comment:
I'm not sure if we should convert the `published` to `Date` and to `IsoString` again. It is already in iso string format and we check it in api. Any reasons to do this double conversion?

Reply:
it is just initialize date type with string, that is what is passed to `createNote`  or we just use published string , ideally this would verify if incoming string is correctly formatted date or not?